### ISSUE TITLE
Request not passed to custom query processors for results queries

### ISSUE THREAD

**Issue by murphyke** [chop-dbhi/serrano#295]
It looks like Avocado's `async_get_result_rows` and `get_result_rows functions` need to be amended to receive `**kwargs`, and Serrano needs to pass request to `async_get_result_rows`.

I can create a PR this morning.

**Comment by bruth** [comment]
Good observation 👍 So there is a silent issue in here. The async workflow enqueues jobs into Redis and does this by pickling the arguments when the job is enqueued (rq does this). Django requests [cannot be pickled](http://stackoverflow.com/a/5823043/407954) so if a request is passed in by Serrano, it will throw an error. Unfortunately this leads to some asymmetry if between the async and non-async endpoints.

**Comment by bruth** [comment]
The alternative is for Serrano to pass in a _description_ of the request for the async workflow, such as headers, `session_key`, `user_id`, etc. so at least some information is available and reconstruct the request object so the processor can use it.

**Comment by murphyke** [comment]
How about a little refactoring of `async_get_result_rows` and `get_result_rows` so that the only thing that the queued job does is `list(rows)`?  Would that work?

**Comment by bruth** [comment]
Good idea.. I am taking a look.

**Comment by murphyke** [comment]
Like:

```
def _evaluate_rows(job_dict):
    """
    Evaluate a Django queryset (job_dict['rows']).

    `job_dict['rows']` is replaced by the result of `list(job_dict['rows'])`
    (i.e. the results of a database query), and `job_dict` is returned.
    """
    job_dict['rows'] = list(job_dict['rows'])
    return job_dict['rows']


def async_get_result_rows(context, view, query_options, job_options=None,
                          **kwargs):
    """
    Creates a new job to asynchronously get result rows and returns the job ID.

    Args:
        See get_result_rows argument list.

    Keyword Arugments:
        job_options: Set as properties on the returned job's meta.
        **kwargs: pass to get_result_rows, ultimately to be passed to
            the get_queryset() method of a QueryProcessor.

    Returns:
        The ID of the created job.
    """
    if not job_options:
        job_options = {}

    queue = get_queue(settings.ASYNC_QUEUE)
    job_dict = get_result_rows(context, view, query_options,
                               evaluate_rows=False, **kwargs)

    job = queue.enqueue(_evaluate_rows, job_dict)
    job.meta.update(job_options)
    job.save()

    return job.id
```

**Comment by bruth** [comment]
Hm, it does not appear the tests pass on Travis. My guess is that that deferred execution does not work well with the queue.

**Comment by murphyke** [comment]
Ah, can't pickle a generator.

**Comment by murphyke** [comment]
I will have to give up on this for today, which means for all next week, since I am on vacation.  /cc @aaron0browne

**Comment by murphyke** [comment]
This was fixed by PR #298
